src/peppr/rotamer_rama.py

```diff
@@ -0,0 +1,1259 @@
+        The rotamer score percentage for the residue, e.g. 0.01 for 1% favored.
+    classification : str
+        The classification of the rotamer score, which can be one of the following:
+        - "FAVORED": The residue is in a favored rotamer conformation.
+        - "ALLOWED": The residue is in an allowed rotamer conformation.
+        - "OUTLIER": The residue is in an outlier rotamer conformation.
+        - "UNKNOWN": The residue's rotamer score could not be determined.
+    error : str | None
+        An error message if any error occurred during rotamer scoring, otherwise None.
+    """
+
+    model_config = ConfigDict(from_attributes=True)
+    model_no: int = Field(
+        description="Model number for the residue, default is 0", default=0
+    )
+    resname: str = Field(description="Residue name")
+    resid: int = Field(description="Residue ID")
+    chain_id: str = Field(description="Chain ID of the residue")
+    observed: dict[str, float] = Field(
+        description="Observed chi angles for the residue, e.g. {'chi1': 60.0, 'chi2': 170.0}"
+    )
+    rotamer_score_pct: float = Field(
+        description="Rotamer score percentage, e.g. 0.01 for 1% favored"
+    )
+    classification: str = Field(
+        description="Rotamer classification: FAVORED, ALLOWED, OUTLIER, UNKNOWN",
+        examples=["FAVORED", "ALLOWED", "OUTLIER", "UNKNOWN"],
+    )
+    error: str | None = Field(
+        description="Error message if any error occurred during rotamer scoring",
+        default=None,
+    )
+
+    @classmethod
+    def from_residue(
+        cls, atom_array: struc.AtomArray, res_id: int, chain_id: str, model_no: int
+    ) -> "ResidueRotamerScore":
+        """
+        Create ResidueRotamerScore from a residue object.
+
+        Parameters
+        ----------
+        atom_array : struc.AtomArray
+            The AtomArray containing the structure.
+        res_id : int
+            The residue ID to check.
+        chain_id : str
+            The chain ID to check.
+        model_no : int
+            The model number to check (usually 0 for single model structures).
+
+        Returns
+        -------
+        ResidueRotamerScore
+            An instance of ResidueRotamerScore containing the rotamer score for the residue.
+        """
+        result = check_rotamer(atom_array, res_id, chain_id, model_no=model_no)
+        return cls(**result)
+
+
+class RotamerScore(BaseModel):
+    """
+    Rotamer score for a given protein structure.
+    This class represents the rotamer scores for all residues in a protein structure,
+    including their names, IDs, observed chi angles, rotamer score percentages, and classifications.
+    """
+
+    model_config = ConfigDict(from_attributes=True)
+    rotamer_scores: list[ResidueRotamerScore] | list[list[ResidueRotamerScore]] = Field(
+        description="List of rotamer scores for each residue in the structure."
+    )
+
+    @classmethod
+    def from_atom_array(
```

Comment:
Why does this exist in addition to `from_atom_array_or_stack()`?

In the underlying logic each model is processed separately right? In this case it will probably do not give measurable speedups when using an `AtomArrayStack` instead of calling this multiple times with each `AtomArray`. Hence, it could simplify the logic, to restrict functions in the module to `AtomArray` as input. In the end the `Evaluator` runs each `AtomArray` individually anyway.